src/Storages/StorageInMemoryMetadata.cpp

```diff
@@ -331,7 +331,7 @@ Block StorageInMemoryMetadata::getSampleBlockForColumns(
     /// Virtual columns must be appended after ordinary, because user can
     /// override them.
     for (const auto & column : virtuals)
-        virtuals_map.emplace(column.name, &column.type);
```

Comment:
The semantic is changed, is it what you expect?

Reply:
I just applied patch from @CurtisJ here